curaPrintTimeEstimator/helpers/ModelTimeCalculator.py

```diff
@@ -1,53 +1,133 @@
+
+    @staticmethod
+    def _findSettings() -> Iterable[Tuple[str, List[str]]]:
+        """
+        Finds the TXT files available in the 'settings' sub folder.
+        :return: An iterable of lists of settings each format: (settings_name, settings_parameters).
+        """
+        directory = "{}/settings".format(Settings.PROJECT_DIR)
+        files = os.listdir(directory)
+        for name in sorted(files):
+            if name.endswith(".txt"):
+                with open("{}/{}".format(directory, name)) as f:
+                    yield name[:-4], f.readlines()
+
+    def gatherPrintTimeData(self, model: str, settings: Dict[str, List[str]],
+                            prev_results: Optional[Dict[str, Dict[str, Optional[int]]]]
+                            ) -> Dict[str, Dict[str, Optional[int]]]:
+        """
+        Gathers data about the estimated print time for one model, all settings and all definitions.
+        :param model: The name of the model file, including the extension.
+        :param settings: A dict with the settings file name and a list of settings for each of the files.
+        :return: A dict with the format {definition: {settings_name: print_time}}.
+        """
+        result = prev_results or {}
+        for definition in self.DEFINITIONS:
+            result.setdefault(definition, {})
+            for setting_name, settings_parameters in settings.items():
+                if result[definition].get(setting_name):
+                    logging.info("Model %s, definition %s and settings %s was already sliced, %s seconds to print.",
+                                 model, definition, settings, result[definition][setting_name])
+                else:
+                    result[definition][setting_name] = self.slice(model, definition, settings_parameters)
+        return result
+
+    def slice(self, model_name: str, definition: str, settings: List[str]) -> Optional[int]:
         """
         Runs the slicer, returning the estimated amount of seconds to print the model.
-        :param model_name: The name of the model, without the .stl extension.
+        :param model_name: The name of the model including the extension.
         :param definition: The definition file to be used, without the .def.json extension.
         :param settings: The extra settings to be passed to the engine.
         :return: The amount of seconds Cura expects the printing will take.
         """
-
         logging.info("Slicing %s with definition %s and settings %s", model_name, definition, settings)
 
         arguments = [
             Settings.CURA_ENGINE,
             "slice", "-v",
             "-o", "NUL" if sys.platform == "win32" else "/dev/null",
-            "-j", "{}/resources/definitions/{}.def.json".format(Settings.CURA_DIR, definition)
+            "-j", "{}/resources/definitions/{}.def.json".format(Settings.CURA_DIR, definition),
+            "-e0", "-l", "{}/models/{}".format(Settings.PROJECT_DIR, model_name)
```

Comment:
The settings must be added as both global and per extruder depending on the setting. That's why I change it first, otherwise the print time is not correct.

Reply:
you're right, it was a mistake when merging